README.md

```diff
@@ -79,6 +100,56 @@ Arguments:
 
 <truncated>
 ```
+    
+# Debugging
+Debugging code requires your current `dotnet new3` session to have its active build session configured to DEBUG, and a debug process from Visual Studio to be attached to the current running `dotnet new3` session. The steps required to accomplish this are outlined below.
+
+## Setup
+
+Start by opening up the **Microsoft.Templating.sln** in Visual Studio. This solution contains the projects needed to run, modify & debug the application. Once your solution is loaded, open up a new command prompt and navigate to the root of your repository. At this point, you can take one of two paths.
+
+Path 1:
+- Run the "setup" script that applies to your environment. 
+- Once complete, type "debug" and press enter.
+  - The "debug" command is an alias that kicks off a command to run a script named **dn3buildmode-debug.cmd**. 
+    - Of which sets the build to a DEBUG mode.
+
+Path 2:
```

Comment:
I'd skip path 1 and just note that if you've already run setup in that console window, you can just type "debug" instead of the whole script name